incubator/exporter-otlp/src/main/java/io/aklivity/zilla/runtime/exporter/otlp/internal/config/OtlpEndpointAdapter.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2021-2023 Aklivity Inc
+ *
+ * Licensed under the Aklivity Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *   https://www.aklivity.io/aklivity-community-license/
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package io.aklivity.zilla.runtime.exporter.otlp.internal.config;
+
+import java.net.URI;
+
+import jakarta.json.Json;
+import jakarta.json.JsonObject;
+import jakarta.json.JsonObjectBuilder;
+import jakarta.json.bind.adapter.JsonbAdapter;
+
+public class OtlpEndpointAdapter implements JsonbAdapter<OtlpEndpointConfig, JsonObject>
+{
+    private static final String PROTOCOL_NAME = "protocol";
+    private static final String LOCATION_NAME = "location";
+    private static final String OVERRIDES_NAME = "overrides";
+
+    private final OtlpOverridesAdapter overrides;
+
+    public OtlpEndpointAdapter()
+    {
+        this.overrides = new OtlpOverridesAdapter();
+    }
+
+    @Override
+    public JsonObject adaptToJson(
+        OtlpEndpointConfig endpoint)
+    {
+        JsonObjectBuilder object = Json.createObjectBuilder();
+        if (endpoint.protocol != null)
+        {
+            object.add(PROTOCOL_NAME, endpoint.protocol);
+        }
+        if (endpoint.location != null)
+        {
+            object.add(LOCATION_NAME, endpoint.location.toString());
+        }
+        if (endpoint.overrides != null)
+        {
+            object.add(OVERRIDES_NAME, overrides.adaptToJson(endpoint.overrides));
+        }
+        return object.build();
+    }
+
+    @Override
+    public OtlpEndpointConfig adaptFromJson(
+        JsonObject object)
+    {
+        String protocol = object.containsKey(PROTOCOL_NAME)
+            ? object.getString(PROTOCOL_NAME)
+            : null;
```

Comment:
If we are defaulting to `http` protocol then we probably should handle it here, and omit the default on output in `adaptToJson(...)` too.

We should also add it as a default in the schema.

Reply:
When you requested to create the `resolve*()` methods in the `OtlpExporterConfig` and do the default handling there, I moved this there, too. Now the `resolveMetrics()`, `resolveSignals()` and `resolveProtocol()` methods are returning the defaults for these in case of the empty value.

```
    private static final String DEFAULT_METRICS_PATH = "/v1/metrics";
    private static final Set<OtlpOptionsConfig.OtlpSignalsConfig> DEFAULT_SIGNALS = Set.of(METRICS);
    private static final String DEFAULT_PROTOCOL = "http";
    private static final long DEFAULT_INTERVAL = Duration.ofSeconds(30).toMillis();
```

I think it makes sense to handle the defaults the same way for all the options for consistency. Pls confirm if you want me to move the default handling back here.

In the schema definition, `http` is the only acceptable option as of now. It's an enum so we can add more options later:
```
                                    "protocol":
                                    {
                                        "enum":
                                        [
                                            "http"
                                        ]
                                    },
```
Pls LMK if you need any changes here.

Reply:
Resolution should be in `OltpExporterConfig` where there is a semantic to follow, like overrides.

However, individual property defaults should be part of parsing the JSON, even in the schema itself if possible, so defaulting of those belong in the parser.

Reply:
got it, fixed